src/hotspots.ts

```diff
@@ -0,0 +1,142 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2021 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import { Commands } from './commands';
+import { computeHotspotContextPanelContent } from './hotspotContextPanel';
+import { HotspotProbability, RemoteHotspot } from './protocol';
+import { resolveExtensionFile } from './util';
+
+export const HOTSPOT_SOURCE = 'SonarQube Security Hotspot';
+
+export const hotspotsCollection = vscode.languages.createDiagnosticCollection('sonarlint-hotspots');
+
+let activeHotspot: RemoteHotspot;
+
+export const showSecurityHotspot = async (hotspot: RemoteHotspot) => {
+  const foundUris = await vscode.workspace.findFiles(`**/${hotspot.filePath}`);
+  if (foundUris.length === 0) {
+    vscode.window.showErrorMessage(`Could not find file '${hotspot.filePath}' in the current workspace.
+
+Please make sure that the right folder is open and bound to the right project on the server,
+ and that the file has not been removed or renamed.`, 'Show Documentation')
+  .then(action => {
+      if (action === 'Show Documentation') {
+        vscode.commands.executeCommand(
+          Commands.OPEN_BROWSER,
+          vscode.Uri.parse('https://docs.sonarqube.org/latest/user-guide/security-hotspots/')
+        );
+      }
+    });
+  } else if (foundUris.length > 1) {
+    // TODO Show quick pick to allow user to select the "right" file
```

Comment:
What about taking the first candidate? In case of duplicate filenames, there is a possibility of duplicate content as well, so this is not very important which one is opened.

Reply:
IMHO this case has a very low probability of happening, so :+1: for using the first one.